sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Expression.scala

```diff
@@ -366,6 +366,8 @@ trait RuntimeReplaceable extends UnaryExpression with Unevaluable {
   // are semantically equal.
   override lazy val preCanonicalized: Expression = child.preCanonicalized
 
+  def isAggregate: Boolean = child.isInstanceOf[AggregateFunction]
```

Comment:
More thoughts about it: technically a `RuntimeReplaceable` can combine one or more expressions, even for something like `f(x) = max(x) - min(x)`.

For aggregate functions, it's very hard to reason about `f(distinct x) FILTER WHERE ...` if `f(x)` combines many expressions, so I think it makes sense to only allow a direct mapping here.

We can make this assumption more explicit here
```
def isAggregate: Boolean = {
  if (child.isInstanceOf[AggregateFunction]) {
    true
  } else {
    assert(child.find(_.isInstanceOf[AggregateFunction]).isEmpty)
    false
  }
}
```